File: Sorting/Insertion Sort/C++/InsertionSort.cpp
// Insertion sort algorithm implemented in C++

#include <iostream>
using namespace std;

// Insertion sort function

void insertionSort(int array[],int size){
	int current;
	int i,j;
	for(i=0;i<size;i++){
		current=array[i];
		for(j=i;j>0 && array[j-1] > current;j--){
			array[j]=array[j-1];
		}
		array[j]=current;
	}
}

// Main function to perform sorting

int main(){
    int i;
    int array_size;
    cout<< "Enter the size of the array to be sorted: ";
    cin>> array_size;
    int array[array_size];
    cout<<"Enter the elements of the array to be sorted: ";
    for(i=0;i<array_size;i++){
        cin>>array[i];
    }
    insertionSort(array,array_size);
    cout<<"Sorted array is:\n";
    for(i=0;i<array_size;i++){
        cout<<array[i]<<"\n";
    }
    return 0;
}

File: Sorting/Insertion Sort/Java/InsertionSort.java
// insertion sort algorithm implemented in java
import java.util.Scanner;
public class InsertionSort {
    /*
    this function takes an array and assumes the first index as sorted part
    takes element from unsorted part and add to sorted part at appropiate  position untill array is not sorted
    */
    public void sort(int[] arr)
    {
        int size=arr.length; 
        for (int i=1;i<size;i++) // starting from index no 1 because it assumes first index as sorted
        {
            int t=i-1;
            int temp=arr[i];
            for (;t>=0&&arr[t]>temp;t--) 
                {
                    arr[t+1]=arr[t];
                }
            arr[t+1]=temp;
        }
    }
    // driver program
     public static void main(String args[])
    {
         Scanner sc=new Scanner(System.in);
         System.out.println("Enter the number of elements in the array: ");
         int n=sc.nextInt();
         System.out.println("Enter "+n+" elements to be soterd in the array: ");
         for(int i=0;i<n;i++)
            int arr[i] =sc.nextInt();
 
        System.out.println("Given Array");
        System.out.print(Arrays.toString(arr));
        InsertionSort ob = new InsertionSort();
        ob.sort(arr);
 
        System.out.println("\nSorted array");
        System.out.print(Arrays.toString(arr));
    }  
}
